ui/components/app/name/name-details/name-display.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { memo } from 'react';
```

Comment:
This is the new component that solves the circular dependency. `name.tsx` and `name-details.tsx` now reference this file, and this file doesn't reference the other two.